templates/cicd/pipeline_cfn.yml

```diff
@@ -222,7 +222,7 @@ Resources:
   BuildTestCommands{{$stage.Name}}:
     Type: AWS::CodeBuild::Project
     Properties:
-      Name: BuildTestCommands-{{$.AppName}}-{{$stage.Name}}
+      Name: BuildTestCommands-{{$.AppName}}-{{$stage.Name}}-{{$.Source.Repository}}-{{$.Source.Branch}}
```

Comment:
Name isn't a required field! https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-project.html#cfn-codebuild-project-name

Can we just remove this field and not deal with the character limit problems :D 

My other question is if we remove it, will existing pipelines still work? Since the change requires a replacement. My guess is yes, but it'd be great to validate it

Reply:
Removed after manual testing. 
I had a pipeline with test commands that had a named CodeBuild project. When I took that line away in the template and ran `pipeline update`, the testcommands stage status changed to `Didn't Run`; when I did `Release Change`, all was well-- each stage succeeded.